### PR TITLE
Add stop on error to the scripts

### PR DESCRIPTION
**Pull request by leshikus**
Motivation: stopping on error is a generic concept in most programming languages which helps to debug

### CODE REVIEW

**Review by pbchekin — APPROVED** [review]
Looks good, one question.

File: scripts/compile-pytorch-ipex.sh [hunk 1]
```diff
@@ -35,7 +37,7 @@ for arg in "$@"; do
 done
 
 set +o xtrace
-if [ -z "$BASE" ]; then
+if [ -z ${BASE:+x} ]; then
```
[the remark this answers is not in the document]
Comment (leshikus) [reply scripts/compile-pytorch-ipex.sh:40]: I used [this ](https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash) for a reference (with a mistake). `-v` seems more compact, I like it

File: scripts/compile-triton.sh
Comment (whitneywhtsang) [inline scripts/compile-triton.sh:1]: Can we allow fails like this?
`ln: failed to create symbolic link 'packages/packages': File exists`

Comment (pbchekin) [reply scripts/compile-triton.sh:1]: This should fix the error:

```
ln -sfT $PACKAGES_DIR $HOME/packages
```

The only problem is when $HOME/packages exists and it is a directory. @leshikus could you please make a PR to add a corresponding check and a fix above?

**Pull request merged**